server/src/main/java/org/apache/cassandra/sidecar/config/yaml/PeriodicTaskConfigurationImpl.java

```diff
@@ -67,47 +71,74 @@ public boolean enabled()
      */
     @Override
-    @JsonProperty("initial_delay_millis")
-    public long initialDelayMillis()
+    @JsonProperty("initial_delay")
+    public MillisecondBoundConfiguration initialDelay()
     {
-        return initialDelayMillis;
+        if (initialDelay == null)
+        {
+            return executeInterval;
+        }
+        return initialDelay;
     }
 
-    @JsonProperty("initial_delay_millis")
-    public void setInitialDelayMillis(long initialDelayMillis)
+    @JsonProperty("initial_delay")
+    public void setInitialDelay(MillisecondBoundConfiguration initialDelay)
     {
-        if (initialDelayMillis > 0)
+        if (initialDelay.compareTo(MillisecondBoundConfigurationImpl.ZERO) > 0)
         {
-            this.initialDelayMillis = initialDelayMillis;
+            this.initialDelay = initialDelay;
         }
         else
         {
-            LOGGER.warn("Invalid initialDelayMillis configuration {}, the minimum value is 0", initialDelayMillis);
-            this.initialDelayMillis = 0;
+            LOGGER.warn("Invalid initialDelay configuration {}, the minimum value is 0", initialDelay);
+            this.initialDelay = MillisecondBoundConfigurationImpl.ZERO;
         }
     }
 
+    @JsonProperty("initial_delay_millis")
+    @Deprecated
+    public void setInitialDelayMillis(long initialDelayMillis)
+    {
+        LOGGER.warn("'initial_delay_millis' is deprecated, use 'initial_delay' instead");
+        setInitialDelay(new MillisecondBoundConfigurationImpl(initialDelayMillis, TimeUnit.MILLISECONDS));
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
-    @JsonProperty("execute_interval_millis")
-    public long executeIntervalMillis()
+    @JsonProperty("execute_interval")
+    public MillisecondBoundConfiguration executeInterval()
     {
-        return executeIntervalMillis;
+        return executeInterval;
     }
 
-    @JsonProperty("execute_interval_millis")
-    public void setExecuteIntervalMillis(long executeIntervalMillis)
+    @JsonProperty("execute_interval")
+    public void setExecuteInterval(MillisecondBoundConfiguration executeInterval)
     {
-        if (executeIntervalMillis > 1)
+        if (executeInterval.compareTo(ONE) > 0)
         {
-            this.executeIntervalMillis = executeIntervalMillis;
+            this.executeInterval = executeInterval;
         }
         else
         {
-            LOGGER.warn("Invalid executeIntervalMillis configuration {}, the minimum value is 1", executeIntervalMillis);
-            this.executeIntervalMillis = 1;
+            LOGGER.warn("Invalid executeInterval configuration {}, the minimum value is 1ms", executeInterval);
+            this.executeInterval = ONE;
         }
     }
+
+    /**
+     * Legacy properties {@code execute_interval_millis}, {@code poll_freq_millis}, and {@code poll_interval_millis}
+     *
+     * @param executeIntervalMillis the interval in milliseconds
+     * @deprecated in favor of {@code execute_interval}
```

Comment:
If we finally decide to leave all the deprecated configs, I think we should have a Jira for removing them.

Reply:
Yeah, I think that should be done after we have the first release though :)